### PR TITLE
Switch to GH-Actions

### PR DESCRIPTION
**Pull request by ixti**
Resolves: #633

### CODE REVIEW

**Comment by ixti** [comment]
Tests are failing due to SSL errors, which are caused by invalid config and/or incorrectly generated certificates. As we do some hackety hackery to mimic different ssl weirdiness, it might be the problem of misconfiguration, but I can't figure out what exactly isn't correct.

**Comment by ixti** [comment]
I tend to disable failing specs temporarily and address those as a separate pr (help is super welcomed).

@tarcieri wdyt?

**Review by tarcieri — APPROVED** [review]

**Review by tarcieri — APPROVED** [review]
There's an open issue for the SSL-related failures: #627. I'm also not sure what's up there... I'm guessing something changed related to either accepted algorithms or hostnames.

Otherwise, this looks good!

File: lib/http/timeout/null.rb [hunk 1]
```diff
@@ -36,8 +36,9 @@ def start_tls(host, ssl_socket_class, ssl_context)
         connect_ssl
 
         return unless ssl_context.verify_mode == OpenSSL::SSL::VERIFY_PEER
+        return unless ssl_context.respond_to?(:verify_hostname) && ssl_context.verify_hostname
```
Comment (tarcieri) [inline lib/http/timeout/null.rb:39]: In hindsight, this should probably have been:

```ruby
return if ssl_context.respond_to?(:verify_hostname) && !ssl_context.verify_hostname
```

...so as to only support explicitly *disabling* hostname verification. As is, it seems hostname verification won't be performed on JRuby at all.

Comment (tarcieri) [reply lib/http/timeout/null.rb:39]: Opened #636 as a follow-up to this comment.

**Comment by ixti** [comment]
Thanks for the fix

**Pull request merged**